bundles/org.openhab.binding.huesync/src/main/java/org/openhab/binding/huesync/internal/discovery/HueSyncDiscoveryParticipant.java

```diff
@@ -0,0 +1,146 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.huesync.internal.discovery;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.jmdns.ServiceInfo;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.huesync.internal.HueSyncConstants;
+import org.openhab.binding.huesync.internal.log.HueSyncLogFactory;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.config.discovery.mdns.MDNSDiscoveryParticipant;
+import org.openhab.core.thing.ThingRegistry;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+
+/**
+ * The {@link HueSyncDiscoveryParticipant} is responsible for discovering
+ * the remote huesync.boxes using mDNS discovery service.
+ *
+ * @author Patrik Gfeller - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = MDNSDiscoveryParticipant.class, configurationPid = "mdnsdiscovery.huesync")
+public class HueSyncDiscoveryParticipant implements MDNSDiscoveryParticipant {
+    private Logger logger = HueSyncLogFactory.getLogger(HueSyncDiscoveryParticipant.class);
+
+    /**
+     *
+     * Match the hostname + identifier of the discovered huesync-box.
+     * Input is like "HueSyncBox-XXXXXXXXXXXX._huesync._tcp.local."
+     * 
+     * @see·<a·href=
+     * "https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=huesync">
+     * Service·Name·and·Transport·Protocol·Port·Number·Registry</a>
+     */
+    private static final String SERVICE_TYPE = "_huesync._tcp.local.";
+
+    private boolean autoDiscoveryEnabled = true;
+
+    protected final ThingRegistry thingRegistry;
+
+    @Activate
+    public HueSyncDiscoveryParticipant(final @Reference ThingRegistry thingRegistry) {
+        this.thingRegistry = thingRegistry;
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Collections.singleton(HueSyncConstants.THING_TYPE_UID);
+    }
+
+    @Override
+    public String getServiceType() {
+        return SERVICE_TYPE;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(ServiceInfo service) {
+        if (this.autoDiscoveryEnabled) {
+            ThingUID uid = getThingUID(service);
+            if (Objects.nonNull(uid)) {
+                try {
```

Comment:
I would prefer simple java, no `Objects` dependency when not needed.
```suggestion
            if (uid != null) {
```